boa/precompile.py

```diff
@@ -0,0 +1,122 @@
+from typing import Optional
+from eth.codecs import abi
+from vyper.address_space import MEMORY
+from vyper.ast import parse_to_ast
+from vyper.codegen.core import IRnode
+from vyper.semantics.types.user import type_from_abi
+from vyper.semantics.types.utils import type_from_annotation
+from vyper.utils import binascii
+from boa.environment import register_precompile
+from vyper.builtins.functions import DISPATCH_TABLE, abi_encode, add_ofst, ir_tuple_from_args, process_inputs
+from vyper.builtins._signatures import BuiltinFunction
+from vyper.codegen.core import IRnode
+from vyper.semantics.types.shortcuts import (
+    UINT256_T,
+)
+
+def extract_arg_types(signature: str):
```

Comment:
this is probably unnecessary if we use `ContractFunctionT.from_FunctionDef` below